File: AtCoder/abc/abc307/b.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(int i = 0; i < (n); ++i)
#define rep1(i, n) for(int i = 1; i < (n); ++i)
using namespace std;
using ll = long long;

bool is_palindrome(string s){
    int n = s.size();

    rep(i, n / 2){
        if(s[i] != s[n-i-1]) return false;
    }
    return true;
}

int main(){
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    int n;
    cin >> n;

    vector<string> s(n);
    rep(i, n) cin >> s[i];

    rep(i, n) rep(j, n){
        if(i == j) continue;

        string cand = s[i] + s[j];

        if(is_palindrome(cand)){
            puts("Yes");
            return 0;
        }
    }

    puts("No");

    return 0;
}

File: AtCoder/abc/abc307/b.py
def main():
    n = int(input())
    s = [input() for _ in range(n)]

    for i in range(n):
        for j in range(n):
            if i == j: continue

            cand = s[i] + s[j]

            if cand == cand[::-1]:
                print("Yes")
                exit()
    
    print("No")

if __name__ == '__main__':
    main()